suite/suite.go

```diff
@@ -23,24 +23,49 @@ type Suite struct {
 	*assert.Assertions
 	require *require.Assertions
 	t       *testing.T
+	f       *testing.F
 }
 
 // T retrieves the current *testing.T context.
 func (suite *Suite) T() *testing.T {
 	return suite.t
 }
 
+// F retrieves the current *testing.F context.
+func (suite *Suite) F() *testing.F {
+	return suite.f
+}
+
 // SetT sets the current *testing.T context.
 func (suite *Suite) SetT(t *testing.T) {
 	suite.t = t
 	suite.Assertions = assert.New(t)
 	suite.require = require.New(t)
 }
 
+// SetF sets the current *testing.F context.
+func (suite *Suite) SetF(f *testing.F) {
+	suite.f = f
+	suite.Assertions = assert.New(f)
+	suite.require = require.New(f)
+}
+
 // Require returns a require context for suite.
 func (suite *Suite) Require() *require.Assertions {
 	if suite.require == nil {
-		suite.require = require.New(suite.T())
+
+		var tT require.TestingT
+		if suite.T() != nil {
+			tT = suite.T()
+		}
+		if suite.F() != nil {
+			tT = suite.F()
+		}
+
+		if tT == nil {
+			panic("Neither T nor F was non-nil")
+		}
```

Comment:
I'm not sure why we want to test for this? Is it not possible for a suite to have both non-nil `T` and `F`?

Reply:
Again, same assumption as above.